README.md

```diff
@@ -1 +1,143 @@
-# x-components
\ No newline at end of file
+# x-components
+
+Micro-frontends help you to use multiple frameworks on your application.
+You can apply several frameworks on one page, per each component.
+
+index.ts
+
+```ts
+import {define} from "@micro-frontends/core";
+import {XLink} from "@micro-frontends/framework";
+import {XContent} from "@micro-frontends/framework";
+
+// define web-components 
+define(XLink);
+define(XContent);
+
+```
+
+index.html
+
+```html
+<main>
+  <navigation>
+    <ul class="margin-top-1em nav flex-column" >
+      <li class="nav-item">
+        <x-link
+            class="nav-link"
+            href="/clock"
+            text="clock (react & svelte)" >
+        </x-link>
+      </li>
+      <li class="nav-item">
+        <x-link
+            class="nav-link"
+            href="/chat"
+            text="chat (svelte + react + vue)" >
+        </x-link>
+      </li>
+      <li class="nav-item">
+        <x-link
+            class="nav-link"
+            href="/todo"
+            text="todo (imba)" >
+        </x-link>
+      </li>
+    </ul>
+  </navigation>
+
+  <section>
+    <x-content id="main" default="main" ></x-content>
+  </section>
+</main>
+
+```
+
+`x-link` router link provide access for `src/pages/` where you can build SPA pages
+`x-content` pages render content
+
+```ts
+// Web component defenition
+import {Component} from "@micro-frontends/core";
+
+// For React and Vue render
+import {renderReact, renderVue} from "@micro-frontends/framework";
+```
+
+Example of the component
+
+React
+
+```ts
+export class XClockReact extends Component {
+    static getName(): string {
+        return 'x-clock-react'
+    }
+
+    render(root: HTMLDivElement) {
+        renderReact(
+            <Clock />,
+            root,
+        )
+    }
+}
+```
+
+Vue
+
+```ts
+export class XChatClientVue extends Component {
+    static getName(): string {
+        return 'x-chat-client-vue'
+    }
+
+    render(root: HTMLDivElement) {
+        renderVue(
+            ChatClient,
+            root,
+        )
+    }
+}
+```
+
+Component with the standart render way
```

Comment:
it would be better to add example of loading component from *.svelte file and mention that this is component using svelte framework